embassy-macros/src/lib.rs

```diff
@@ -280,3 +280,69 @@ pub fn main(args: TokenStream, item: TokenStream) -> TokenStream {
     };
     result.into()
 }
+
+#[cfg(not(any(feature = "nrf", feature = "stm32", feature = "rp")))]
```

Comment:
This will cause the `std` macro to be enabled if the user is using `embassy` directly on a not supported chip (not nrf, stm32, rp).

Instead, could you add a `std` feature and enable it from `embassy-std`, like this? https://github.com/embassy-rs/embassy/blob/master/embassy-nrf/Cargo.toml#L23